### PR TITLE
feat(ME-1206): add ssh auth proxy and aws session manager support

### PR DESCRIPTION
**Pull request by btoonk**
# Description

add ssh auth proxy and aws session manager support.
Fixes # ME-1206

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested locally.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: internal/ssh/Shellsession.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	retryParams     retry.RepeatableExponentialRetryer
+	sshChannel      ssh.Channel
+	sshWindowHeight int
+	sshWindowWidth  int
+	ssmWindowHeight int
+	ssmWindowWidth  int
+}
+
+func (s *ShellSession) Initialize(log log.T, sessionVar *ShellSession) {
+	// s = sessionVar
```
Comment (pedroandrade) [inline internal/ssh/Shellsession.go:32]: nit: remove the code comment

File: internal/connector/discover/ec2.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go-v2/aws"
```
Comment (pedroandrade) [inline internal/connector/discover/ec2.go:10]: nice, we also should remove from the go.mod

File: internal/ssh/Shellsession.go (hunk 1 above)
Comment (btoonk) [reply internal/ssh/Shellsession.go]: done

File: internal/connector/discover/ec2.go (hunk 2 above)
Comment (btoonk) [reply internal/connector/discover/ec2.go:10]: it's still in use in other files.

File: internal/ssh/Shellsession.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
+package ssh
```
[the remark this answers is not in the document]
Comment (btoonk) [reply internal/ssh/Shellsession.go]: done

**Review by th3wingman — APPROVED** [review]
tested in staging and lab, all core features work as expected

**Pull request merged**